src/Test/QuickCheck/Laws/Data/Enum.purs

```diff
@@ -0,0 +1,43 @@
+
+module Test.QuickCheck.Laws.Data.Enum where
+
+import Prelude
+import Control.Monad.Eff.Console (log)
+import Data.Enum (pred, succ, class Enum)
+import Data.Maybe (Maybe(Nothing))
+import Test.QuickCheck (QC, quickCheck')
+import Test.QuickCheck.Arbitrary (class Arbitrary)
+import Type.Proxy (Proxy)
+
+checkEnum
+  ∷ ∀ eff a
+  . (Arbitrary a, Enum a, Ord a)
+  ⇒ Proxy a
+  → QC eff Unit
+checkEnum _ = do
+
+  log "Checking 'GT ordering' law for Enum"
+  quickCheck' 1000 gtordering
+
+  log "Checking 'LT ordering' law for Enum"
+  quickCheck' 1000 ltordering
+
+  log "Checking 'predsuccpred' law for BoundedEnum"
+  quickCheck' 1000 predsuccpredLaw
+
+  log "Checking 'succpredsucc' law for BoundedEnum"
+  quickCheck' 1000 succpredsuccLaw
+  
+  where
+    gtordering ∷ a → Boolean
+    gtordering a = succ a == Nothing || succ a > pred a
+
+    ltordering ∷ a → Boolean
+    ltordering a = succ a == Nothing || pred a < succ a
+  
+    predsuccpredLaw :: a -> Boolean
+    predsuccpredLaw a = succ a == Nothing || (pred a >>= succ >>= pred) == pred a
```

Comment:
At first I thought this might be a typo and we should be checking `pred a == Nothing` instead of `succ a == Nothing`, but actually I think we can remove that check altogether:

If `pred a` is `Nothing`, then `(pred a >>= succ >>= pred) == pred a` will always be true, regardless of whether you have a law-abiding Enum instance. If `pred a` is `Just something`, then whether `(pred a >>= succ >>= pred) == pred a` holds depends on whether you have a law-abiding Enum instance.